### PR TITLE
Fix invalid adjoint names for subroutine and module (closes #1717)

### LINKED ISSUE #1717: invalid adjoint kernel module names
adjoint kernel module names are currently the tangent linear name with `_adj` appended. This does not conform to the `_mod _code _type` LFRic convention. The type and code names are also not updated. `psyclone-kern -gen stub xxx.F90` complains about this.

### PR DESCRIPTION
**Pull request by rupertford**
prepend `adj_` to tangent-linear module and subroutine names rather than appending as appending breaks the LFRic rules about ending in `_mod`, `_code`, etc. Also removes prepended `tl_` if it exists in the original tangent linear name. Metadata names (metadata `_type` and reference to code are not updated as this is waiting on proper parsing of kernel metadata).

### CODE REVIEW

**Comment by rupertford** [comment]
Ready for first review from @arporter

**Review by arporter — CHANGES_REQUESTED** [review]
All good apart from one minor typo.
Tests and examples are all OK with compilation.
Only omission is documentation. I've checked and actually we say nothing about the naming of the adjointed code and I think we should - just a couple of sentences should do it.

File: src/psyclone/psyad/tl2ad.py [hunk 1]
```diff
@@ -57,15 +57,35 @@
 #: the generated test-harness code.
 #: TODO #1346 this tolerance should be user configurable.
 INNER_PRODUCT_TOLERANCE = 1500.0
-#: The suffix we will prepend to a routine and container name when generating
-#: its adjoint.
-ADJOINT_NAME_SUFFIX = "_adj"
+#: The prefix we will prepend to a routine, container and metadata
+#: names when generating : the adjoint. If the original name contains
```
[the remark this answers is not in the document]
Comment (rupertford) [reply src/psyclone/psyad/tl2ad.py:61]: Oops.

**Comment by rupertford** [comment]
Added a section about naming.

**Comment by rupertford** [comment]
Ready for next review from @arporter

**Review by arporter — APPROVED** [review]
All requested changes have been made.
Doc updated and builds fine.
All examples and tests OK with compilation.

**Pull request merged**